backend-stack-bubbling/app.py

```diff
@@ -572,6 +580,47 @@ def post():
                 "is_upvote": info["is_upvote"]
             }), 200)
 
+class DeclareBestAnswer(Resource):
+    @staticmethod
+    @jwt_required()
+    def post():
+        info = BestAnswerInfo.parse_args()
+        identity = get_jwt_identity()
+        responseMessage = ""
+        currentUser = UserCollection.find_one(
+            {
+                "email": identity["email"]
+            })
+        questionID = uuid.UUID(info["question_id"])
+        answerID = uuid.UUID(info["answer_id"])
+        if currentUser is not None:
+            answer = QuestionCollection.find_one(
+                {
+                    "_id": questionID,
+                    "answers.is_best_answer": True
+                })
+            if answer is None:
+                QuestionCollection.update(
+                {
+                    "_id" : questionID,
+                    "answers._id": answerID
+                },
+                {
+                    "$set":  
+                    {
+                        "answers.$.is_best_answer": True
+                    }
+                })
+                responseMessage = "Best Answer Declared!"
+                return make_response(jsonify(responseMessage), 201)
+            else:
```

Comment:
Here when I tried to declare an answer to be the best answer, 
and this answer is already the best answer (meaning this answer's is_beast_answer = true), 
it should have the return message of "Best Answer id already declared!", 
but I got no return message

Reply:
But you can't declare an answer to be best answer if it's already best answer because the best answer button does not appear for a best answer.

Reply:
Could you clarify this further?

Reply:
Ah I see, probably the line 
"answers.is_best_answer": True
was missing and it crashed my program.
Now its fine, good work